protos/flyteidl/service/admin.proto

```diff
@@ -313,6 +313,16 @@ service AdminService {
     };
   }
 
+  rpc UpdateProject (flyteidl.admin.ProjectUpdateRequest) returns (flyteidl.admin.ProjectUpdateResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/projects"
```

Comment:
If it's an update, can we use `put` or `patch` instead of `post`? `post`ing to this endpoint should have the result of creating a new project.